presto-verifier/src/main/java/com/facebook/presto/verifier/framework/LimitQueryDeterminismAnalysis.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.verifier.framework;
+
+public enum LimitQueryDeterminismAnalysis
```

Comment:
@caithagoras What is `Limit` here stands for? Why not just `QueryDeterminismAnalysis` or better yet `QueryDeterminismAnalysisResult`?

Reply:
We do an additional determinism check for queries with `LIMIT` clause on the outer layer if all previous determinism analysis runs say `DETERMINISTIC`, and that's the `LimitQueryDeterminismAnalysis`.

Select/Insert/CreateTableAsSelect queries with `LIMIT` clause at the same level of the select query are eligible, expect for queries with `ORDER BY` clause at the same level.
- `SELECT ... LIMIT ...`
- `SELECT ... UNION ALL SELECT ... LIMIT ...`
- `INSERT INTO ... SELECT ... LIMIT ...`
- `CREATE TABLE ... AS SELECT ... LIMIT ...`

What we do:
- We run a `count(1)` query with the LIMIT clause removed to get the `total row count`.
- If `total row count` > `limit count`, treat as non-deterministic.

Reply:
@caithagoras Thanks for explaining? Would it add this to the docs if it is not there already?

Reply:
Will add this to a separate PR: https://github.com/prestodb/presto/pull/12686

Reply:
@caithagoras That PR seems to be stuck for a few months now. What's blocking it?

Reply:
I'll need to address the comments. There is no blocker.